### PR TITLE
Fix @index not displaying when using #key for iteration

### PR DESCRIPTION
**Pull request by garrettreed**
For #70 

I will make a similar fix for https://github.com/canjs/canjs/issues/2361 when this is approved.

### CODE REVIEW

File: src/mustache_core.js [hunk 1]
```diff
@@ -32,12 +32,18 @@ var attr = require("can-util/dom/attr/attr");
...
 var mustacheLineBreakRegExp = /(?:(?:^|(\r?)\n)(\s*)(\{\{([^\}]*)\}\}\}?)([^\S\n\r]*)($|\r?\n))|(\{\{([^\}]*)\}\}\}?)/g,
 	// A helper for calling the truthy subsection for each item in a list and putting them in a document Fragment.
-	getItemsFragContent = function(items, isObserveList, helperOptions, options){
-		var frag = getDocument().createDocumentFragment();
-		for (var i = 0, len = items.length; i < len; i++) {
-			append(frag, helperOptions.fn( isObserveList ? items.attr('' + i) : items[i], options) );
-		}
-		return frag;
+	getItemsFragContent = function(items, helperOptions, scope) {
+        var result = [],
```
Comment (justinbmeyer) [inline src/mustache_core.js]: please use tabs

File: src/mustache_core.js [hunk 2]
```diff
@@ -32,12 +32,18 @@ var attr = require("can-util/dom/attr/attr");
...
+            i = 0,
+            len = 0;
+        for (i = 0, len = items.length; i < len; i++) {
```
Comment (justinbmeyer) [inline src/mustache_core.js]: does `.attr('length')` need to be called if a `can.Map`?

File: src/mustache_core.js [hunk 3]
```diff
@@ -170,8 +176,11 @@ var core = {
 					var isObserveList = types.isMapLike(finalValue);
 
 					if(isObserveList ? finalValue.attr("length") : finalValue.length) {
```
Comment (justinbmeyer) [inline src/mustache_core.js]: looks like `.attr("length")` might get called here ... but I think this should probably be moved into the helper.

**Comment by garrettreed** [comment]
@justinbmeyer the changes you requested are ready for review.

File: src/utils.js [hunk 4]
```diff
@@ -67,15 +68,38 @@ module.exports = {
...
+	getItemsFragContent: function(items, helperOptions, scope, asVariable) {
+		var result = [],
+			len = observationReader.get(items, 'length');
+
+		for (var i = 0; i < len; i++) {
+			var aliases = {
+				"%index": i,
+				"@index": i
+			};
+			var item = observationReader.get(items, '' + i);
```
Comment (justinbmeyer) [inline src/utils.js]: What happens if you do

```
var item = compute(items, i);
```

**Pull request merged**